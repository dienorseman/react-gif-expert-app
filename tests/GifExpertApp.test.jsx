const { render, screen, waitFor } = require("@testing-library/react")
const { GiffExpertApp } = require("../src/GiffExpertApp")

describe('Test on <GifExpertApp />', () => {
    test('should match the snapshot', () => {
        const { container } = render( <GiffExpertApp /> );
        expect( container ).toMatchSnapshot();
        // screen.debug();
    })
    test('should change state', async () => {
        render( <GiffExpertApp />)
        const loadStatus = screen.getByText( 'Loading...' )
        expect( loadStatus ).toBeTruthy()
        console.log(loadStatus );
        await waitFor(
            () => expect( loadStatus ).toBeFalsy()
        )
        screen.debug()
    })
})