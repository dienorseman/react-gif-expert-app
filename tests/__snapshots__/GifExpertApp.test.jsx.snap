// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifExpertApp /> should match the snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    death note
  </h3>
  <h2>
    Loading...
  </h2>
  <ol
    class="card-grid"
  />
</div>
`;
